lib/slim-rails.rb

```diff
@@ -1,18 +1,15 @@
 require 'slim-rails/version'
 require 'rails'
 require 'slim'
+require 'slim-rails/register_engine'
 
 module Slim
   module Rails
     class Railtie < ::Rails::Railtie
       config.app_generators.template_engine :slim
 
       initializer 'slim_rails.configure_template_digestor' do |app|
-        if config.respond_to?(:assets)
-          config.assets.configure do |env|
-            env.assets.register_engine '.slim', Slim::Template
-          end
-        end
+        Slim::Rails::RegisterEngine.register_engine(app, config)
```

Comment:
I extracted the register_engine logic to `Slim::Rails::RegisterEngine.register_engine`.